include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1032,52 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Compute a collection of bounding boxes so that all cells for which the given predicate is true, are
+   * completely enclosed in at least one of the bounding boxes.
+   *
+   * For each cell at a given refinement level containing active cells for which predicate is true,
+   * the function creates a bounding box of its children for which predicate is true.
+   *
+   * This results in a cover of these cells; the parameters @p allow_merge and @p max_boxes are used to
+   * reduce the number of cells at a computational cost and covering a bigger n-dimensional volume.
+   *
+   * The parameters to control the algorithm are:
+   * - @p predicate : the property of the cells to enclose e.g. IteratorFilters::LocallyOwnedCell .
+   *  The predicate is tested only on active cells.
+   * - @p refinement_level : it defines the level at which the initial bounding box are created. The refinement
+   *  should be set to a coarse refinement level. If @p refinement_level is higher than the number of levels of the
+   *  triangulation an exception is thrown.
+   * - @p allow_merge : This flag allows for box merging and, by default, is false. The algorithm has a cost of
+   *  O(N^2) where N is the number of the bounding boxes created from the refinement level; for this reason, if
+   *  the flag is set to true, make sure to choose wisely a coarse enough @p refinement_level.
+   * - @p max_boxes : the maximum number of bounding boxes to compute. If more are created the smaller ones are
+   *  merged with neighbors. By default after merging the boxes which can be expressed as a single one no
+   *  more boxes are merged. See the BoundingBox::get_neighbor_type () function for details.
+   *  Notice only neighboring cells are merged (see the @p get_neighbor_type  function in bounding box class): if
+   *  the target number of bounding boxes max_boxes can't be reached by merging neighbors an exception is thrown
```

Comment:
why is it not possible to always take the union of all bounding boxes?

Reply:
Because usually if b1, b2 are bounding boxes, then b1 union b2 (the union is done thinking them as closed sets in the space) is strictly contained inside b1 merged to b2.
The NeighbourType tries to avoid making a very awful union. For instance
|A|0|
|0|A|
Imagine the A are two bounding boxes (the 0 are owned by someone else): I don't want to merge the two A as it would result into a bounding box of the whole space.

Reply:
okay, so when creating the boxes you are fine with an inexact cover, but when merging you disallow a change of the covered area?